workflow/util/util.go

```diff
@@ -811,7 +811,9 @@ func FormulateRetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSucce
 	switch wf.Status.Phase {
 	case wfv1.WorkflowFailed, wfv1.WorkflowError:
 	default:
-		return nil, nil, errors.Errorf(errors.CodeBadRequest, "workflow must be Failed/Error to retry")
```

Comment:
And then perhaps a different error if you attempt to restart a running workflow. (If something catches that before here, I can't see it).

Reply:
@Joibel  - what about pending / suspended workflows?

Reply:
I believe so. Previously if you got here with a pending/suspended, attempting a retry, you'd get a failure telling you it should have failed/errored. You're not intentionally changing that behaviour, so best tell people the conditions where retry is allowed.

Reply:
Fixed... it now explicitly mentions `wfv1.WorkflowSucceeded` case, and then adds default behavior.
Thanks, it makes more sense now.